### PR TITLE
Support for osfamily RedHat

### PR DESCRIPTION
**Pull request by slauger**
Hi,

i've added support for the osfamily RedHat (CentOS, RedHat). For CentOS 7 it was required to replace the Jinja function `equalto` with `match`. The function `equalto` is not yet available in the pyrhon-jinja2 package shipped with CentOS 7 (currently v2.7.2). 

See also https://github.com/openshift/openshift-ansible/issues/3463

Cheers, Simon

### CODE REVIEW

**Comment by slauger** [comment]
@klen are you still actively maintaining this repository? There are a lot of open PRs like this. 

You currently managing a lot of ansible modules. I can imagine that this might a bit too much work for a single person.

**Comment by nkakouros** [comment]
Hello @slauger ,

I was recently given commit permissions on this repo and I plan to look into some PRs soon (ie in a few days)!

**Comment by slauger** [comment]
Hi @tterranigma,

would be great! This role has already 175 forks and the number is constantly increasing. If you need help just let me know.

File: templates/client.conf.j2 [hunk 1]
```diff
@@ -42,10 +42,10 @@ persist-tun
 {{ openvpn_ca_file_contents }}
 </ca>
 <cert>
-{{ openvpn_client_cert_output |default([{'item':client,'stdout':''}])|selectattr('item', 'equalto', client)|map(attribute='stdout')|list|first }}
+{{ openvpn_client_cert_output |default([{'item':client,'stdout':''}])|selectattr('item', 'match', client)|map(attribute='stdout')|list|first }}
```
Comment (nkakouros) [inline templates/client.conf.j2:45]: Can you explain why this is needed?

File: tasks/install.yum.yml [hunk 2]
```diff
@@ -1,3 +1,28 @@
...
+    name: epel-release
+  when: openvpn_use_external_repo
+
+- name: Install requirements (RedHat)
+  apt: name={{item}} force=yes
+  with_items: [openvpn, openssl, zip]
+
+- name: Install dependencies (RedHat)
+  apt: name={{item}} force=yes
+  when: openvpn_use_pam_users|default(false)
```
Comment (nkakouros) [inline tasks/install.yum.yml]: I wonder,

```suggestion
  when: openvpn_use_pam_users|length
```

won't do the same thing? I know this is the way it is in `install.deb.yml` too, but let's discuss overall improvements.

**Review by nkakouros — CHANGES_REQUESTED** [review]

**Comment by nkakouros** [comment]
In `install.yum.yml`, there are many `apt` tasks. Could you make them into yum tasks?

File: templates/client.conf.j2 (hunk 1 above)
Comment (slauger) [reply templates/client.conf.j2:45]: Have a look on my comment in the PR. CentOS 7 (and RedHat as well) are shipping an older version of python-jinja2.

File: tasks/install.yum.yml (hunk 2 above)
Comment (slauger) [reply tasks/install.yum.yml]: You're right... maybe i overseen that openvpn_use_pam_users is an list, not an dict.

Comment (slauger) [reply tasks/install.yum.yml]: You're right... maybe i overseen that openvpn_use_pam_users is an list, not an dict.

**Comment by nkakouros** [comment]
BTW, there is also #53 that takes care of bridging. Would you mind merging that branch into your branch so that we benefit from the work there and the author there takes credit too?

I have no experience with Red Hat but I think we should setup travis to test the role on it (or CentOS) to be sure that it works as expected.

**Comment by Adirael** [comment]
Hello,

I've been trying out this PR on RHEL and it's having trouble installing EPEL there, since there's not an epel-release package. 

I suggest doing something similar to what https://github.com/geerlingguy/ansible-role-repo-epel/blob/master/tasks/main.yml does. 

I forked @slauger's fork and changed this. Tested on RHEL and CentOS.

**Comment by slauger** [comment]
Hi @Adirael,

thanks. Seems like that `epel-release` is only available via `centos-7-extras`. I could not find a package for RedHat and OracleLinux.

An simple solution would be to set the default for `openvpn_use_external_repo` on RHEL and OracleLinux to `false` or add a dependency to geerlingguy.repo-epel. What do you think?

I merged the changes from #53 but didn't had the time to test all the changes yet. It could be that there are still some problems with the code.

**Comment by Adirael** [comment]
My use case would prefer EPEL, so I would opt for having geerlingguy.repo-epel as dependency or re-creating that here, it comes out as +22 lines (https://github.com/Adirael/Stouts.openvpn/commit/df800670d65b983fd79baeb9029077e238ff61f8)

That fork has #105 and #53. I've only played with it for a little while, but it looks alright :)

**Comment by slauger** [comment]
I added a dependency for osfamily RedHat in `meta/main.yml` and successfully deployed a new OpenVPN server with the following inventory settings.

```
openvpn_host: "{{inventory_hostname}}"
openvpn_port: 8443
openvpn_proto: tcp
openvpn_key_size: 4096
openvpn_tls_auth: true
openvpn_cipher: AES-256-GCM
openvpn_server_options:
  - 'tls-version-min 1.2'
openvpn_clients:
  - foobar
openvpn_use_pam: no
openvpn_unified_client_profiles: yes
```

I hope everything is fine now.

**Comment by slauger** [comment]
Any update on this?

**Review by nkakouros — APPROVED** [review]

**Comment by nkakouros** [comment]
@slauger As I wrote elsewhere, I will go ahead and merge this without automated testing for the platform it adds support for. Tests for redhat will be added later.

**Pull request merged**